core/shared/src/test/scala/cats/parse/ParserTest.scala

```diff
@@ -1229,20 +1249,37 @@ class ParserTest extends munit.ScalaCheckSuite {
     }
   }
 
-  property("rep0Sep with unit sep is the same as rep0") {
-    forAll(ParserGen.gen, Gen.choose(0, Int.MaxValue), Arbitrary.arbitrary[String]) {
-      (genP, min0, str) =>
-        val min = min0 & Int.MaxValue
-        val p1a = Parser.rep0Sep(genP.fa, min = min, sep = Parser.unit)
-        val p1b = genP.fa.rep0(min = min)
+  property("repSep0 with unit sep is the same as rep0") {
+
+    val minMax =
+      for {
+        min <- biasSmall(0)
+        max <- biasSmall(Integer.max(min, 1))
+      } yield (min, max)
 
-        assertEquals(p1a.parse(str), p1b.parse(str))
+    forAll(ParserGen.gen, biasSmall(0), Arbitrary.arbitrary[String]) { (genP, min, str) =>
+      val p1a = Parser.repSep0(genP.fa, min = min, sep = Parser.unit)
+      val p1b = genP.fa.rep0(min = min)
+
+      assertEquals(p1a.parse(str), p1b.parse(str))
 
-        val min1 = if (min < 1) 1 else min
-        val p2a = Parser.repSep(genP.fa, min = min1, sep = Parser.unit)
-        val p2b = genP.fa.rep(min = min1)
+      val min1 = if (min < 1) 1 else min
+      val p2a = Parser.repSep(genP.fa, min = min1, sep = Parser.unit)
+      val p2b = genP.fa.rep(min = min1)
 
-        assertEquals(p2a.parse(str), p2b.parse(str))
+      assertEquals(p2a.parse(str), p2b.parse(str))
+    } &&
```

Comment:
note the `&&`, without this, the first test does not run.

forAll returns a value, not a side effect.

Reply:
nice catch!